src/nameck.rs

```diff
@@ -63,7 +63,7 @@ use std::u32;
 ///
 /// [INC]: https://github.com/sorear/smetamath-rs/issues/11
 #[derive(Copy, Clone, Debug, PartialOrd, Ord, PartialEq, Eq, Default, Hash)]
-pub struct Atom(u32);
+pub struct Atom(pub u32);
```

Comment:
What is the motivation for making this public? This will make garbage atoms user-constructible, and its relation to the other invariants of the library are not clear. I don't see why you would ever want to be able to write `Atom(37)`. Read only access to the number can be done by a `Atom::index()` function.

Reply:
If the resolver needs to create `Symbol`s, then it should have access to a name generator which can ensure that the `Atom`s are globally unique, and it should not be directly responsible for manipulating the number.

Reply:
See [Atom generator, avoid exposing atom's internal](https://github.com/david-a-wheeler/metamath-knife/pull/89/commits/900899718b3b45f7844d40d39573559798c48459)